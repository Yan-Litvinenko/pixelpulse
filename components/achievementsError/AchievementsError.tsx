import React from 'react';
import styles from '@/styles/components/achievementsError/AchievementsError.module.scss';
// import { useModal } from '../../hooks/useModal';

export default function AchievementsError(): React.JSX.Element {
    // const modal = useModal('social');

    return (
        <ul className={styles.error}>
            <li className={styles.error__item}>
                Check your internet connection. Ensure that you are connected to a stable network.
            </li>
            <li className={styles.error__item}>
                Try refreshing the page by pressing Ctrl+R or Command+R on your keyboard.
            </li>
            <li className={styles.error__item}>If you are using a VPN, try disabling it and reloading the page.</li>
            <li className={styles.error__item}>
                Clear your browser&prime;s cache and cookies, then try reloading the page.
            </li>
            <li className={styles.error__item}>
                If the error persists, please write to{' '}
                <span onClick={() => {} /*  modal.open()*/} data-testid="achievements-modal-error">
                    me
                </span>
            </li>
        </ul>
    );
}
