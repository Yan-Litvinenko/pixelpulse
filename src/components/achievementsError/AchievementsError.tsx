import React from 'react';
import styles from './AchievementsError.module.scss';
import { modalOpenHandler } from '../../store/slices/modalSlice';
import { soundsModalTrigger } from '../../store/slices/soundsSlice';
import { useDispatch } from 'react-redux';
import type { AppDispatch } from '../../store/store';

const AchievementsError = (): React.JSX.Element => {
    const dispatch = useDispatch<AppDispatch>();

    return (
        <ul className={styles.error} data-testid="achievements-list-error">
            <li className={styles.error__item}>
                Check your internet connection. Ensure that you are connected to a stable network.
            </li>
            <li className={styles.error__item}>
                Try refreshing the page by pressing Ctrl+R or Command+R on your keyboard.
            </li>
            <li className={styles.error__item}>If you are using a VPN, try disabling it and reloading the page.</li>
            <li className={styles.error__item}>Clear your browser's cache and cookies, then try reloading the page.</li>
            <li className={styles.error__item}>
                If the error persists, please write to{' '}
                <span
                    onClick={() => {
                        dispatch(soundsModalTrigger());
                        dispatch(modalOpenHandler({ key: 'social' }));
                    }}
                    data-testid="achievements-modal-error"
                >
                    me
                </span>
            </li>
        </ul>
    );
};

export { AchievementsError };
