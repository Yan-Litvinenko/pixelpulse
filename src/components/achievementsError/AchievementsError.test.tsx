import React from 'react';
import { AchievementsError } from './AchievementsError';
import { render, screen } from '@testing-library/react';
import * as reduxHooks from 'react-redux';

jest.mock('react-redux');

const mockDispatch = jest.spyOn(reduxHooks, 'useDispatch');

describe('AchievementsError component', (): void => {
    test('Render', (): void => {
        const component = mockDispatch.mockReturnValue(jest.fn());
        render(<AchievementsError />);

        const error = screen.getByTestId('achievements-error');
        expect(error).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });
});
