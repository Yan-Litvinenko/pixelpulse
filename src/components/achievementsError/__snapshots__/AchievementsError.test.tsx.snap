// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementsError component Render 1`] = `
[MockFunction] {
  "calls": [
    [],
    [],
  ],
  "results": [
    {
      "type": "return",
      "value": [MockFunction],
    },
    {
      "type": "return",
      "value": [MockFunction],
    },
  ],
}
`;
